airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py

```diff
@@ -299,7 +299,13 @@ def download_data(self, url: str, chunk_size: int = 1024) -> tuple[str, str]:
             tmp_file, "wb"
         ) as data_file:
             for chunk in response.iter_content(chunk_size=chunk_size):
-                data_file.write(decompressor.decompress(chunk))
+                try:
+                    # see if it's compressed. we are seeing some that are not all of a sudden.
+                    # but let's also guard against the case where sendgrid changes it back.
+                    data_file.write(decompressor.decompress(chunk))
+                except zlib.error as e:
+                    # it's not actually compressed!
```

Comment:
Should we log a warning here to inform that an error has occurred?

Reply:
For better or worse, this is the expected case I've seen at the moment. I would be curious how often it work and it doesn't work. What's the right way to log so I can see (datadog?) but not have it fill up the user-visible logs. Is that a thing?